Binaries not at correct location?

Hey 👋, 
I just wanted to compile the project after a fresh clone. The quickstart guide suggested running `cargo run --bin rustflags` but that doesn't work for me, as Cargo is not able to find the binaries. I checked the docs on this and I think the binaries are in an incorrect location. According to the [docs](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#binaries) it should be in `src/bin` or defined in the Cargo.toml. Am I doing something wrong here?

Thanks, that makes it clear, but the script doesn't seem to work for me:
```
❯ ./bin/rustflags.rs
warning: `package.edition` is unspecified, defaulting to `2021`
   Compiling rustflags v0.0.0 (/home/piet/projects/rustc_codegen_clr/bin)
error[E0583]: file not found for module `build_backend`
 --> rustflags.rs:2:1
  |
2 | mod build_backend;
  | ^^^^^^^^^^^^^^^^^^
  |
  = help: to create the module `build_backend`, create file "build_backend.rs" or "build_backend/mod.rs"
  = note: if there is a `mod build_backend` elsewhere in the crate already, import it with `use crate::...` instead

error[E0432]: unresolved import `crate::build_backend::ilasm_check`
 --> rustflags.rs:3:5
  |
3 | use crate::build_backend::ilasm_check;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `ilasm_check` in `build_backend`

error[E0425]: cannot find function `cargo_build_env` in module `crate::build_backend`
 --> rustflags.rs:5:43
  |
5 |     let build_env = crate::build_backend::cargo_build_env();
  |                                           ^^^^^^^^^^^^^^^ not found in `crate::build_backend`

Some errors have detailed explanations: E0425, E0432, E0583.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `rustflags` (bin "rustflags") due to 3 previous errors
```

> Thanks, that makes it clear, but the script doesn't seem to work for me:
> 
> ```
> ❯ ./bin/rustflags.rs
> warning: `package.edition` is unspecified, defaulting to `2021`
>    Compiling rustflags v0.0.0 (/home/piet/projects/rustc_codegen_clr/bin)
> error[E0583]: file not found for module `build_backend`
>  --> rustflags.rs:2:1
>   |
> 2 | mod build_backend;
>   | ^^^^^^^^^^^^^^^^^^
>   |
>   = help: to create the module `build_backend`, create file "build_backend.rs" or "build_backend/mod.rs"
>   = note: if there is a `mod build_backend` elsewhere in the crate already, import it with `use crate::...` instead
> 
> error[E0432]: unresolved import `crate::build_backend::ilasm_check`
>  --> rustflags.rs:3:5
>   |
> 3 | use crate::build_backend::ilasm_check;
>   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `ilasm_check` in `build_backend`
> 
> error[E0425]: cannot find function `cargo_build_env` in module `crate::build_backend`
>  --> rustflags.rs:5:43
>   |
> 5 |     let build_env = crate::build_backend::cargo_build_env();
>   |                                           ^^^^^^^^^^^^^^^ not found in `crate::build_backend`
> 
> Some errors have detailed explanations: E0425, E0432, E0583.
> For more information about an error, try `rustc --explain E0425`.
> error: could not compile `rustflags` (bin "rustflags") due to 3 previous errors
> ```
 I forgot to include one of the files(`build_backend.rs`). Should be fixed now.

--- a/bin/build_backend.rs
+++ b/bin/build_backend.rs
@@ -72,7 +72,7 @@ pub fn cargo_build_env() -> String {
 pub fn ilasm_check() {
     match std::process::Command::new(&*ILASM_PATH).output(){
         Ok(_)=>println!("An CIL assembler has been detected."),
-        Err(err)=>panic!("Could not find the CIL assembler at name/path {:?}, due to {err:?}. 
+        Err(err)=>panic!("Could not find the CIL assembler at name/path {:?}, due to {err:?}.
 Please instal the CIL assembler, and/or set the ILASM_PATH enviroment variable to point to your CIL assembler.",*ILASM_PATH)
     }
 }
@@ -97,7 +97,9 @@ pub fn absolute_backend_path() -> PathBuf {
         } else if cfg!(target_os = "windows") {
             std::fs::canonicalize("../target/debug/rustc_codegen_clr.dll").unwrap()
         } else if cfg!(target_os = "macos") {
-            std::fs::canonicalize("../target/debug/librustc_codegen_clr.dylib").unwrap()
+            let current_dir = std::env::current_dir().unwrap();
+            let dylib_path = current_dir.join("target/debug/librustc_codegen_clr.dylib");
+            std::fs::canonicalize(dylib_path).expect("Could not find the backend dylib")
         } else {
             panic!("Unsupported target OS");
         }
@@ -106,7 +108,9 @@ pub fn absolute_backend_path() -> PathBuf {
     } else if cfg!(target_os = "windows") {
         std::fs::canonicalize("../target/release/rustc_codegen_clr.dll").unwrap()
     } else if cfg!(target_os = "macos") {
-        std::fs::canonicalize("../target/release/librustc_codegen_clr.dylib").unwrap()
+        let current_dir = std::env::current_dir().unwrap();
+        let dylib_path = current_dir.join("target/release/librustc_codegen_clr.dylib");
+        std::fs::canonicalize(dylib_path).expect("Could not find the backend dylib")
     } else {
         panic!("Unsupported target OS");
     }
@@ -156,7 +160,9 @@ static RUSTC_CODEGEN_CLR_LINKER: std::sync::LazyLock<PathBuf> = std::sync::LazyL
             .unwrap();
         //TODO: Fix this for other platforms
         if cfg!(target_os = "linux") || cfg!(target_os = "macos") {
-            std::fs::canonicalize("../target/debug/linker").unwrap()
+            let current_dir = std::env::current_dir().unwrap();
+            let linker_path = current_dir.join("target/debug/linker");
+            std::fs::canonicalize(linker_path).unwrap()
         } else if cfg!(target_os = "windows") {
             std::fs::canonicalize("../target/debug/linker.exe").unwrap()
         } else {
@@ -169,7 +175,9 @@ static RUSTC_CODEGEN_CLR_LINKER: std::sync::LazyLock<PathBuf> = std::sync::LazyL
             .unwrap();
         //TODO: Fix this for other platforms
         if cfg!(target_os = "linux") || cfg!(target_os = "macos") {
-            std::fs::canonicalize("../target/release/linker").unwrap()
+            let current_dir = std::env::current_dir().unwrap();
+            let linker_path = current_dir.join("target/release/linker");
+            std::fs::canonicalize(linker_path).unwrap()
         } else if cfg!(target_os = "windows") {
             std::fs::canonicalize("../target/release/linker.exe").unwrap()
         } else {
@@ -187,13 +195,14 @@ fn build_backend() -> Result<(), String> {
         .args(["build", "--lib", "--release"])
         .output()
         .map_err(|err| err.to_string())?;
+    let current_dir = std::env::current_dir().unwrap();
     let _out = std::process::Command::new("cargo")
-        .current_dir("../cilly")
+        .current_dir(current_dir.join("cilly"))
         .args(["build", "--bin", "linker"])
         .output()
         .expect("could not build the backend");
     let _out = std::process::Command::new("cargo")
-        .current_dir("../cilly")
+        .current_dir(current_dir.join("cilly"))
         .args(["build", "--bin", "linker", "--release"])
         .output()
         .expect("could not build the backend");

